multiraft/transport.go

```diff
@@ -100,6 +103,18 @@ func NewLocalRPCTransport() Transport {
 	}
 }
 
+// EnableEvents creates the Events channel. All future clients will send an
+// *interceptMessage to that channel and block until the message is
+// acknowledged. EnableEvents should be called before any clients are created,
+// i.e. prior to calls to Connect.
+func (lt *localRPCTransport) EnableEvents() {
```

Comment:
My concern is mainly with the dramatically different implementation (including a change to blocking behavior) depending on whether EnableEvents has been called. Since EnableEvents can only be called immediately after creation it's effectively a different type anyway.